File: result/translate/chatgpt/java2python/python/MINIMUM_STEPS_MINIMIZE_N_PER_GIVEN_CONDITION.py
def f_gold(n):
    table = [0] * (n + 1)
    
    for i in range(n + 1):
        table[i] = n - i
    
    for i in range(n, 0, -1):
        if i % 2 == 0:
            table[i // 2] = min(table[i] + 1, table[i // 2])
        if i % 3 == 0:
            table[i // 3] = min(table[i] + 1, table[i // 3])
    
    return table[1]

File: result/translate/starchat/java2cpp/c++/MINIMUM_STEPS_MINIMIZE_N_PER_GIVEN_CONDITION.cpp
int f_gold ( int n ) {
  vector <int> table ( n + 1, n );
  for ( int i = 0; i <= n; i++ ) {
    for ( int j = i; j > 0; j /= 2 ) {
      if (! (j % 2 > 0)) table[j] = min ( table[j], table[j - 1] + 1 );
      if (! (j % 3 > 0)) table[j] = min ( table[j], table[j - 1] + 1 );
    }
  }
  return table[1];
}